src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -152,6 +152,7 @@ public class MqttClient implements Closeable {
     private final CallbackEventManager callbackEventManager = new CallbackEventManager();
     private final Spool spool;
     private final ExecutorService executorService;
+    private SecurityService securityService;
```

Comment:
should be final.

Reply:
adding the final keyword, the IDE complains that it might not be initialized.

This is caused by the call to the protected constructor. Additonally the constructor used for the spooler unit / integration tests does have SecurityService as a parameter.

Reply:
Yes exactly, you're not initializing it which is why I left the comment. You need to make sure it is always initialized or don't have it as a field if it is not going to be initialized by some code paths.